### PR TITLE
Closes #314 | Add dataloader for Indonesia chinese mt robust eval

### LINKED ISSUE #314: Create dataset loader for Indonesia-Chinese-MTRobustEval
Dataloader name: `indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?indonesia_chinese_mtrobusteval

|     Dataset| indonesia_chinese_mtrobusteval |
|-------------|---|
| Description | The dataset is curated for the purpose of evaluating the robustness of Neural Machine Translation (NMT) towards natural occuring noise (typo, slang, code switching, etc.). The dataset is crawled from Twitter, then pre-processed to obtain sentences with noise. The dataset consists of a thousand noisy sentences. The dataset is translated into Chinese manually as the benchmark for evaluating the robustness of NMT. |
| Subsets | - |
| Languages | ind, cmn |
| Tasks | Machine Translation |
| License | MIT (mit) |
| Homepage | https://github.com/supryzhu/Indonesia-Chinese-MTRobustEval |
| HF URL | - |
| Paper URL | - |

### PR DESCRIPTION
**Pull request by TysonYu**
Please name your PR after the issue it closes. You can use the following line: "Closes #314" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by jen-santoso — CHANGES_REQUESTED** [review]
Hello @TysonYu 
Thank you for the dataloader implementation.
The dataloader runs well, however, there are some nits to fix as commented.

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            for line in values:
+                writer.write(line)
+# This template is based on the following template from the datasets package:
+# https://github.com/huggingface/datasets/blob/master/templates/new_dataset_script.py
+
+
+# This allows you to run your dataloader with `python [dataset_name].py` during development
+# TODO: Remove this before making your PR
+if __name__ == "__main__":
+    datasets.load_dataset(__file__)
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:182]: Done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  year      = {2023},
+  url       = {https://github.com/supryzhu/Indonesia-Chinese-MTRobustEval},
+  doi       = {None},
+  biburl    = {None},
+  bibsource = {None}
+}
+"""
+
+
+_DATASETNAME = "[indonesia_chinese_mtrobusteval]"
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:48]: Done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+_DESCRIPTION = """\
+The dataset is curated for the purpose of evaluating the robustness of Neural Machine Translation (NMT) towards natural occuring noise
+(typo, slang, code switching, etc.). The dataset is crawled from Twitter, then pre-processed to obtain sentences with noise.
+The dataset consists of a thousand noisy sentences. The dataset is translated into Chinese manually as the benchmark for evaluating the robustness of NMT.
+"""
+
+_HOMEPAGE = "https://github.com/supryzhu/Indonesia-Chinese-MTRobustEval"
+
+# TODO: Add languages related to this dataset
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:58]: Done, removed this line

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The dataset is curated for the purpose of evaluating the robustness of Neural Machine Translation (NMT) towards natural occuring noise
+(typo, slang, code switching, etc.). The dataset is crawled from Twitter, then pre-processed to obtain sentences with noise.
+The dataset consists of a thousand noisy sentences. The dataset is translated into Chinese manually as the benchmark for evaluating the robustness of NMT.
+"""
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:20]: Done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+                name=datasets.Split.TRAIN,
+                # Whatever you put in gen_kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": file_path+'.jsonl',
+                    "split": "train",
+                },
+            )
+        ]
+
+    # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:146]: Done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    _DATASETNAME: "https://github.com/supryzhu/Indonesia-Chinese-MTRobustEval/raw/main/data/Indonesia-Chinese.xlsx",
+}
+
+_SUPPORTED_TASKS = [Tasks.MACHINE_TRANSLATION]  # example: [Tasks.TRANSLATION, Tasks.NAMED_ENTITY_RECOGNITION, Tasks.RELATION_EXTRACTION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+class NewDataset(datasets.GeneratorBasedBuilder):
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:76]: Done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    """The dataset consists of a thousand noisy sentences. The dataset is translated into Chinese manually as the benchmark for evaluating the robustness of NMT."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="indonesia_chinese_mtrobusteval_source",
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:84]: done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+            version=SOURCE_VERSION,
+            description="indonesia_chinese_mtrobusteval source schema",
+            schema="source",
+            subset_id="indonesia_chinese_mtrobusteval",
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:88]: done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+        ),
+        SEACrowdConfig(
+            name="indonesia_chinese_mtrobusteval_seacrowd_t2t",
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:91]: done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+            version=SEACROWD_VERSION,
+            description="indonesia_chinese_mtrobusteval SEACrowd schema",
+            schema="seacrowd_t2t",
+            subset_id="indonesia_chinese_mtrobusteval",
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:95]: done

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "indonesia_chinese_mtrobusteval_source"
```
[the remark this answers is not in the document]
Comment (TysonYu) [reply seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:99]: done

**Comment by jen-santoso** [comment]
@TysonYu 
Also, could you upload `__init__.py` as well?

**Comment by TysonYu** [comment]
> @TysonYu Also, could you upload `__init__.py` as well?

Sure. Done~

File: seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+                    i += 1
+
+    def _write_jsonl(self, filepath, values):
+        with jsonlines.open(filepath, "w") as writer:
+            for line in values:
+                writer.write(line)
+
+
+# This template is based on the following template from the datasets package:
+# https://github.com/huggingface/datasets/blob/master/templates/new_dataset_script.py
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/indonesia_chinese_mtrobusteval/indonesia_chinese_mtrobusteval.py:154]:
```suggestion
```

**Comment by jen-santoso** [comment]
@TysonYu Thank you for the updates! now it's LGTM. Merging...

**Pull request merged**